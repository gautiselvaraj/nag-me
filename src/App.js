import React from 'react';
import styled, {ThemeProvider} from 'styled-components';
import theme from './theme'
import './global-styles';
import Header from './components/Header';
import SearchSort from './components/SearchSort';
import Nags from './components/Nags';
import NagForm from './components/NagForm';

const App = styled.div`
  font-family: 'Open Sans', sans-serif;
  line-height: 1.35;
  width: 350px;
`;

const nags = [
  {
    id: 1,
    title: 'Drink Water',
    nextNag: Date.parse(new Date('2017-12-4')),
    repeats: '2 min',
    paused: false,
    nagCount: 0,
    createdAt: Date.parse(new Date('2017-12-2')),
  },
  {
    id: 2,
    title: 'Take a deep breath',
    nextNag: Date.parse(new Date('2017-12-5')),
    repeats: '20 min',
    paused: false,
    nagCount: 0,
    createdAt: Date.parse(new Date('2017-12-2')),
  },
  {
    id: 3,
    title: 'Take a short walk',
    nextNag: Date.parse(new Date('2017-12-25')),
    repeats: '5 hour',
    paused: true,
    nagCount: 0,
    createdAt: Date.parse(new Date('2017-12-2')),
  },
  {
    id: 4,
    title: 'Sleep once in a while',
    nextNag: Date.parse(new Date('2018-2-5')),
    repeats: false,
    paused: false,
    nagCount: 0,
    createdAt: Date.parse(new Date('2017-12-2')),
  }
]

export default () => (
  <ThemeProvider theme={theme}>
    <App>
      <Header />
      <SearchSort />
      <Nags nags={nags} />
      <NagForm />
    </App>
  </ThemeProvider>
);
