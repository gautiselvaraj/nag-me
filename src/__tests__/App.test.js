import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { App } from '../App';
import 'jest-styled-components';

describe('<App />', () => {
  it('should match snapshot', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <App nagInit={jest.fn()} activePage="Index" />
    );
    expect(tree).toMatchSnapshot();
  });
});
