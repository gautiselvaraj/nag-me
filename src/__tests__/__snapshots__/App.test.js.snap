// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "black": "#111111",
      "grey": "#898989",
      "greyDark": "#777777",
      "greyDarker": "#595959",
      "greyDarkest": "#222",
      "greyLight": "#a5a5a5",
      "greyLighter": "#dddddd",
      "greyLightest": "#efefef",
      "main": "#214154",
      "mainDarken": "#08283B",
      "mainLighten": "#3B5B6E",
      "white": "#f6f6f6",
    }
  }
>
  <styled.div>
    <Connect(Header) />
    <TransitionMotion
      didLeave={[Function]}
      styles={
        Array [
          Object {
            "key": "Index",
            "style": Object {
              "translateX": Object {
                "damping": 26,
                "precision": 0.01,
                "stiffness": 350,
                "val": 0,
              },
            },
          },
        ]
      }
      willEnter={[Function]}
      willLeave={[Function]}
    >
      [Function]
    </TransitionMotion>
  </styled.div>
</ThemeProvider>
`;
